File: frontend/src/api/clients.ts
import api from "@/lib/api"
import { ApiResponse } from "@/types/api"
import { Client } from "@/types/client"

export const clientsApi = {
  list: async (orgId: string): Promise<ApiResponse<Client[]>> => {
    const res = await api.get(`/clients/${orgId}`)
    return res.data
  },
  get: async (clientId: string): Promise<ApiResponse<Client>> => {
    const res = await api.get(`/clients/detail/${clientId}`)
    return res.data
  },
  create: async (orgId: string, name: string, description?: string): Promise<ApiResponse<Client>> => {
    const res = await api.post("/clients/", { organization_id: orgId, name, description })
    return res.data
  },
  update: async (clientId: string, data: { name?: string; description?: string; status?: string }): Promise<ApiResponse<Client>> => {
    const res = await api.put(`/clients/${clientId}`, data)
    return res.data
  },
  delete: async (clientId: string): Promise<ApiResponse<{ message: string }>> => {
    const res = await api.delete(`/clients/${clientId}`)
    return res.data
  },
}

File: backend/app/api/v1/clients.py
from typing import List
from uuid import UUID
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession
from app.api.deps import get_current_user, get_db
from app.models.user import User
from app.schemas.client import ClientResponse, CreateClientRequest, UpdateClientRequest
from app.schemas.common import SuccessResponse
from app.services.client_service import ClientService

router = APIRouter()

@router.post("/", response_model=SuccessResponse[ClientResponse])
async def create_client(data: CreateClientRequest, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    return SuccessResponse(data=await ClientService(db).create_client(data, current_user.id))

@router.get("/{org_id}", response_model=SuccessResponse[List[ClientResponse]])
async def list_clients(org_id: UUID, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    return SuccessResponse(data=await ClientService(db).list_clients(org_id, current_user.id))

@router.get("/detail/{client_id}", response_model=SuccessResponse[ClientResponse])
async def get_client(client_id: UUID, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    return SuccessResponse(data=await ClientService(db).get_client(client_id, current_user.id))

@router.put("/{client_id}", response_model=SuccessResponse[ClientResponse])
async def update_client(client_id: UUID, data: UpdateClientRequest, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    return SuccessResponse(data=await ClientService(db).update_client(client_id, data, current_user.id))

@router.delete("/{client_id}")
async def delete_client(client_id: UUID, db: AsyncSession = Depends(get_db), current_user: User = Depends(get_current_user)):
    await ClientService(db).delete_client(client_id, current_user.id)
    return SuccessResponse(data={"message": "Client deleted"})
